src/gluonts/torch/model/mqf2/module.py

```diff
@@ -0,0 +1,259 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from typing import List, Optional, Tuple
+
+import torch
+
+from gluonts.core.component import validated
+from gluonts.torch.model.deepar.module import DeepARModel
+from gluonts.torch.modules.distribution_output import DistributionOutput
+
+from cpflows.flows import ActNorm
+from cpflows.icnn import PICNN
+from .icnn_utils import DeepConvexNet, SequentialNet
+
+
+class MQF2MultiHorizonModel(DeepARModel):
+    @validated()
+    def __init__(
+        self,
+        freq: str,
+        context_length: int,
+        prediction_length: int,
+        num_feat_dynamic_real: int,
+        num_feat_static_real: int,
+        num_feat_static_cat: int,
+        cardinality: List[int],
+        distr_output: DistributionOutput,
```

Comment:
It looks like this *need* to be to a `MQF2DistributionOutput`, right? If so, then there's no need to have this input at construction time, and can just be constructed and passed to the `super` constructor I think